moveit_planners/ompl/ompl_interface/test/test_state_validity_checker.cpp

```diff
@@ -0,0 +1,361 @@
+  {
+    ASSERT_NE(planning_context_, nullptr) << "Initialize planning context before adding path constraints.";
+
+    // set the robot to a known position that is withing the joint limits and collision free
+    robot_state_->setJointGroupPositions(joint_model_group_, position_in_joint_limits);
+
+    // create position constraints around the given robot state
+    moveit_msgs::Constraints path_constraints;
+    Eigen::Isometry3d ee_pose = robot_state_->getGlobalLinkTransform(ee_link_name_);
+    path_constraints.name = "test_position_constraints";
+    path_constraints.position_constraints.push_back(createPositionConstraint(
+        { ee_pose.translation().x(), ee_pose.translation().y(), ee_pose.translation().z() }, { 0.1, 0.1, 0.1 }));
+
+    moveit_msgs::MoveItErrorCodes error_code_not_used;
+    bool succeeded = planning_context_->setPathConstraints(path_constraints, &error_code_not_used);
+    ASSERT_TRUE(succeeded) << "Failed to set path constraints on the planning context";
+
+    auto checker = std::make_shared<ompl_interface::StateValidityChecker>(planning_context_.get());
+    checker->setVerbose(VERBOSE);
+
+    // use a scoped OMPL state so we don't have to call allocState and freeState
+    // (as recommended in the OMPL documantion)
+    ompl::base::ScopedState<> ompl_state(state_space_);
+    state_space_->copyToOMPLState(ompl_state.get(), *robot_state_);
+
+    if (VERBOSE)
+      ROS_INFO_STREAM(ompl_state.reals());
+
+    bool result = checker->isValid(ompl_state.get());
+    EXPECT_TRUE(result);
+
+    // move the position constraints away from the curren end-effector position to make it fail
+    moveit_msgs::Constraints path_constraints_2(path_constraints);
+    path_constraints_2.position_constraints.at(0).constraint_region.primitive_poses.at(0).position.z += 0.2;
+    // path_constraints_2.name = "test_position_constraints";
+    // path_constraints_2.position_constraints.push_back(createPositionConstraint(
+    //     { ee_pose.translation().x(), ee_pose.translation().y(), ee_pose.translation().z() + 0.2 }, { 0.1, 0.1, 0.1
+    //     }));
+
+    succeeded = planning_context_->setPathConstraints(path_constraints_2, &error_code_not_used);
+    ASSERT_TRUE(succeeded) << "Failed to set path constraints on the planning context";
+
+    // checker = std::make_shared<ompl_interface::StateValidityChecker>(planning_context_.get());
+    // checker->setVerbose(VERBOSE);
+
+    // clear the cached validity of the state before checking again,
+    // otherwise the path constraints will not be checked.
+    ompl_state->as<ompl_interface::JointModelStateSpace::StateType>()->clearKnownInformation();
+
+    bool result_2 = checker->isValid(ompl_state.get());
+    EXPECT_FALSE(result_2);
+  }
+
+  // /***************************************************************************
+  //  * END Test implementation
+  //  * ************************************************************************/
```

Comment:
use one or the other style of comments, both seems odd

Reply:
done

Reply:
Done